windowing.py

```diff
@@ -0,0 +1,127 @@
+    #			  the windows
+    #
+    # Input: dict_matrix = dict
+    # Input: rows = int (number of rows in a trace)
+    # Input: window_size = int (# of samples within one window when creating the windows)
+    # Input: stride = int (# of samples between the start of each window)
+    #
+    # Return: windowed_data = np array (np array of the windowed data)
+    x = dict_matrix['x']
+    y = dict_matrix['y']
+    z = dict_matrix['z']
+    nanos = dict_matrix['nanos']
+    np_data = np.array([x, y, z, nanos], 'd')
+    windowed_data = view_as_windows(np_data, (rows, window_size), step=stride)
+    return windowed_data
+
+
+def find_initial_lift_times(lift_windows, rows):
+    # Description: returns the start and end times from lift_windows np array
+    #
+    # Input: lift_windows = np array
+    # Input: rows = int (number of rows in a trace)
+    #
+    # Returns: int start (start time in nanos)
+    #		   int end (end time in nanos)
+
+    start = lift_windows[0][rows-1][0]
+    end = lift_windows[-1][-1][-1]
+    return start, end
+
+
+def find_lift_windows(windowed_data, divisor):
+    # Description: takes the windowed data and calculates variance for each window.
+    #				Then calculates variance threshold and returns an np array of
+    #				the windows that meet the variance threshold.
+    #
+    # Input: windowed_data = np array
+    # Input: divisor = int (will use divisor to calculate variance threshold so
+    #						that the threshold is a function of the max variance.
+    #						Threshold = (max variance)/divisor)
+    #
+    # Return: lift_windows = np array
+
+    windows = windowed_data[0]
+    windows_no_nanos = np.delete(windows, 3, 1)
+    variances = np.var(windows_no_nanos, axis=2, ddof=1)
+    var_sum = np.sum(variances, axis=1)
+    indices = np.argwhere(var_sum > max(var_sum)/6)
+    if (len(indices) <= 1):
+        indices = indices[0]
+    else:
+        indices = np.squeeze(indices)
+        count = indices[0]-1
+        for i in range(len(indices)):
+            count += 1
+            if count != indices[i]:
+                end = i
+                break
+        indices = indices[:i]
+    lift_windows = np.take(windows, indices, axis=0)
+    return lift_windows
+
+
+def refine_start_time(windowed_data, threshold):
+    # Description: finds the more refined start time by calcualting variacne
+    #				on the smaller windows and returning the start time of the first
+    #				window that meets the thresholdusing a smaller threshold.
+    #
+    # Input: windowed_data = np array
+    # Input: threshold = int (variance threshold for start of lift
+    #
+    #
+    # Return: refined start time = int (in nanos)
+    windows = windowed_data[0]
+    windows_no_nanos = np.delete(windowed_data[0], 3, 1)
+    variances = np.var(windows_no_nanos, axis=2, ddof=1)
+    var_sum = np.sum(variances, axis=1)
+    indexes = np.argwhere(var_sum > threshold)
+    if (len(indexes) < 1):
```

Comment:
You can also use indexes.size == 0 to check if a numpy array is empty. Has the benefit of not having to apply an external function to the data structure and make it a bit more intuitive than < 1.